### PR TITLE
FIX: respect nofollow exclusion setting in topic featured links.

### PR DESCRIPTION
**Pull request by vinothkannans**
Previously, nofollow attribute is not removed even when a domain is added to the `exclude_rel_nofollow_domains` site setting.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/render-topic-featured-link.js [hunk 1]
```diff
@@ -11,6 +11,16 @@ export function addFeaturedLinkMetaDecorator(decorator) {
 export function extractLinkMeta(topic) {
   const href = topic.get("featured_link");
   const target = User.currentProp("external_links_in_new_tab") ? "_blank" : "";
+  const domain = topic.get("featured_link_root_domain");
+  let allowList = Discourse.SiteSettings.exclude_rel_nofollow_domains;
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/lib/render-topic-featured-link.js:15]: We have to move away from `Discourse.SiteSettings` constant usage as it's a global variable.

Instead could you pass in `allowList` as an option to this function? See this PR didn't add this but the same approach should be done for `User.currentProp` - we could pass that in as an option `externalLinksInNewTab: true` when calling the function.

This is necessary for newer versions of Ember and to allow us to upgrade our application.

**Comment by SamSaffron** [comment]
@vinothkannans can you try removing the `Discourse.SiteSettings` call this week so we can merge?

**Comment by vinothkannans** [comment]
Sorry for the delay. It's now updated.
Since the site settings are now available in all the rest models I just retrieved it from the topic argument.

File: app/assets/javascripts/discourse/app/lib/render-topic-featured-link.js (hunk 1 above)
Comment (vinothkannans) [reply app/assets/javascripts/discourse/app/lib/render-topic-featured-link.js]: > See this PR didn't add this but the same approach should be done for `User.currentProp`

If I update it then I have to change 3 other files which are not related to this PR. So I will do it in a separate PR.

**Review by eviltrout — APPROVED** [review]

**Pull request merged**